### PR TITLE
#170947667 notification for new request made

### PR DESCRIPTION
**Pull request by Kagorora**
#### What does this PR do?
- Notify the manager when a new request is made.
#### Description of Task to be completed?
- for the successful creation of a request, an email is sent to the intended manager
- for the successful creation of a request, an in-app notification is sent to the intended manager
#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git checkout ft-notification-for-new-Request-170947667 ` to switch to this branch.
- run `npm install` to install dependencies.
- go to this link to see live notification `https://codepen.io/king_Bomaye/pen/VwLbQGX?editors=1111`
- create a .env file then add environment variable as shown in ` .env.example`,  check our slack channel `code-knights-12` for .env

```
   MANAGER_PSWD=$2b$10$SO7hwwylEnHKZIv68w/ePOK4yybPnhXa5.tQSUBe.PSb6FBKyuVO.
```

- create two Postgres database for development and then test
- run `npm run createTables` to create tables for you

- run `npm run dev-start` to start the server.

- login using admin credentials POST `http://localhost:4000/api/v1/auth/login`

`{
    "email": "william.ishimwe@andela.com",
    "password": "Password@1"
}`

- copy the `manager's token` in the following app `https://codepen.io/king_Bomaye/pen/VwLbQGX?editors=1111`

Screen Shot 2020-02-28 at 20 48 29

- signup a regular user user by POST `localhost:4000/api/v1/auth/signup`
{
                 "firstName": "Mugenzi",
                  "lastName": "Jean",
                  "gender": "male",
                  "passportNumber": "ui8908dv",
                 "email": "david@andela.com",
                 "password": "Niyonkuru@1"
            }

- make a new request by POST `http://localhost:4000/api/v1/trips/returnTrip`

{
	"origin": "Burundo",
	"destination": "kgl",
	"departureDate": "9002-04-02",
	"returnDate": "9003-05-18",
	"accommodation": "XYZ campus",
	"reason": "Having a meeting",
	"passportNumber": "yuyut56"
}

- will see a successful message for success request and an email will be sent to the manager 

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?

- [#170947667](https://www.pivotaltracker.com/story/show/170947667)

#### Screenshots (if appropriate)
              

Successfully created notification(I have to create a small UI to demonstrate successful notification)

Screen Shot 2020-02-27 at 16 12 27

 Email notification send to the Manager(William in this case is the manager)

Screen Shot 2020-02-27 at 11 24 07

#### Questions:
N/A

### CODE REVIEW

**Comment by Niyonsengaeric** [comment]
@Kagorora why should a user provide his passport number while he has provided it during the signup Process

**Comment by Niyonsengaeric** [comment]
Screen Shot 2020-02-29 at 11 43 06

it is not working on my side

**Comment by Kagorora** [comment]
> @Kagorora why should a user provide his passport number while he has provided it during the signup Process

> @Niyonsengaeric You are very right, There was no need for that, So I have removed it

**Comment by Kagorora** [comment]
> Screen Shot 2020-02-29 at 11 43 06
> 
> it is not working on my side

>Please recheck, I have corrected the error

File: package.json [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
-    "test": "NODE_ENV=test npm run dropTables && NODE_ENV=test sequelize db:migrate && NODE_ENV=test sequelize db:seed:all && NODE_ENV=test mocha --require @babel/polyfill --require @babel/register ./src/tests/mainTest.test.js",
+    "test": "NODE_ENV=test npm run dropTables && NODE_ENV=test sequelize db:migrate && NODE_ENV=test sequelize db:seed:all && NODE_ENV=test mocha --require @babel/polyfill --require @babel/register ./src/tests/mainTest.test.js --timeout 60000 --exit",
     "coverage": "nyc npm test && nyc report --reporter=text-lcov | coveralls",
     "dropTables": "sequelize db:migrate:undo:all",
     "createTables": "sequelize db:migrate && sequelize db:seed:all",
     "build": "babel src --out-dir dist --source-maps inline --copy-files",
-    "start": "nodemon dist/index.js",
-    "dev-start": "nodemon --exec babel-node src/index.js"
+    "start": "nodemon dist/app.js",
+    "dev-start": "nodemon --exec babel-node src/app.js"
```
Comment (T2Wil) [inline package.json:14]: It seems like code in index.js was moved to app.js and hence changed the `dev-start` script.
@Kagorora , is there any reason?

Comment (Kagorora) [reply package.json:14]: Yes, I did change our entry point, from index.js to app.js  as  I needed socket.io to listen on the same port as  our server, as a consequence I have changed our script to start the server and deleted index.js as it was no longer used anywhere,

**Review by Niyonsengaeric — APPROVED** [review]

**Review by T2Wil — APPROVED** [review]

File: src/db/migrations/20200228100755-create-notification.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/db/migrations/20200228100755-create-notification.js]: please check this.

File: src/db/models/notification.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      allowNull: false,
+      validate: {
+        isIn: {
+          args: [['new_request', 'request_approved_or_rejected', 'edited_request', 'new_comment']],
+          msg: 'Invalid value',
+        },
+      },
+    },
+  }, {});
+  Notification.associate = function(models) {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/db/models/notification.js]: please check this.

**Review by kagabof — APPROVED** [review]
@Kagorora, you did a great job on this Pull request. please keep pushing.

**Pull request merged**